mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -63,7 +63,18 @@ class HashingTF(override val uid: String)
     "This is useful for discrete probabilistic models that model binary events rather " +
     "than integer counts")
 
-  setDefault(numFeatures -> (1 << 18), binary -> false)
+  /**
+   * The hash algorithm used when mapping term to integer.
+   * Supported options: "murmur3" and "native".
+   * (Default = "murmur3")
+   * @group param
+   */
+  val hashAlgorithm = new Param[String](this, "hashAlgorithm", "The hash algorithm used when " +
+    "mapping term to integer. Supported options: murmur3(default) and native.",
```

Comment:
Instead of hard coding the supported options you could use `${feature.HashingTF.supportedHashAlgorithms.mkString(",")}`. This is done in other params, like [here](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala#L192).